diff --git a/configure b/configure
index cc6a06f..a788963 100755
--- a/configure
+++ b/configure
@@ -4313,7 +4313,7 @@ IFS="- /.
 
   case $host_alias in
   *aix*)
-    APXS_LIBEXECDIR=`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-brtl -Wl,-bI:$APXS_LIBEXECDIR/httpd.exp"
     PHP_AIX_LDFLAGS="-Wl,-brtl"
     build_type=shared
@@ -4508,7 +4508,7 @@ EOF
   if test "$?" != "0"; then
     APACHE_INSTALL="$APXS -i -a -n php5 $SAPI_SHARED" # Old apxs does not have -S option
   else 
-    APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     if test -z `$APXS -q SYSCONFDIR`; then
       APACHE_INSTALL="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                        $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -5656,7 +5656,7 @@ IFS="- /.
     { echo "configure: error: Please note that Apache version >= 2.0.40 is required" 1>&2; exit 1; }
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -6504,7 +6504,7 @@ IFS="- /.
     { echo "configure: error: Please note that Apache version >= 2.0.44 is required" 1>&2; exit 1; }
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -7339,7 +7339,7 @@ IFS="- /.
 
   case $host_alias in
   *aix*)
-    APXS_LIBEXECDIR=`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-brtl -Wl,-bI:$APXS_LIBEXECDIR/httpd.exp"
     PHP_AIX_LDFLAGS="-Wl,-brtl"
     build_type=shared
@@ -7534,7 +7534,7 @@ EOF
   if test "$?" != "0"; then
     APACHE_HOOKS_INSTALL="$APXS -i -a -n php5 $SAPI_SHARED" # Old apxs does not have -S option
   else 
-    APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     if test -z `$APXS -q SYSCONFDIR`; then
       APACHE_HOOKS_INSTALL="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                        $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
diff --git a/sapi/apache2filter/config.m4 b/sapi/apache2filter/config.m4
index 2c8ff58..a5648a3 100644
--- a/sapi/apache2filter/config.m4
+++ b/sapi/apache2filter/config.m4
@@ -68,7 +68,7 @@ if test "$PHP_APXS2FILTER" != "no"; then
     AC_MSG_ERROR([Please note that Apache version >= 2.0.40 is required])
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index 6238470..00dcf39 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -67,7 +67,7 @@ if test "$PHP_APXS2" != "no"; then
     AC_MSG_ERROR([Please note that Apache version >= 2.0.44 is required])
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
